jdbc/src/java/org/apache/hive/jdbc/HiveConnection.java

```diff
@@ -813,6 +818,22 @@ RegistryBuilder.<ConnectionSocketFactory> create().register("https", socketFacto
     return httpClientBuilder.build();
   }
 
+  private boolean isRequestTrackingEnabled() {
+    return Boolean.valueOf(sessConfMap.get(JdbcConnectionParams.JDBC_PARAM_REQUEST_TRACK));
+  }
+
+  private Supplier<String> getSessionId() {
```

Comment:
I just realized that this could potentially be simplified to avoid using a `Supplier`, e.g. `requestInterceptor.setSessionId(getSessionId())`, where `getSessionId()` immediately does the concatenation and returns `String`. Is the logic behind a `Supplier` because it's potentially expensive for users who don't want to enable request tracking? If so, then it would be good to add a comment explaining that.

Reply:
Supplier because in the very-first request (when the interceptor is created), there is no session and hence sessionId at all, shall I clarify with a comment?

Reply:
Oh, I see now. Thank you! Yes, I think a comment would help.

Reply:
makes sense, added a comment